packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/IntentSender.java

```diff
@@ -158,6 +169,45 @@ Intent buildIntent(
         intent.setPackage(null);
       }
     }
+    Boolean needIgnore = ignoredPackages != null && ignoredPackages.size() > 0;
+    if (needIgnore || showChooser) {
+      PackageManager packageManager;
+      if (activity != null) {
+        packageManager = activity.getPackageManager();
+      } else {
+        packageManager = applicationContext.getPackageManager();
+      }
+      List<ResolveInfo> activities = packageManager.queryIntentActivities(intent, 0);
+      ArrayList<Intent> targetIntents = new ArrayList<Intent>();
+
+      if (needIgnore) {
+        for (ResolveInfo currentInfo : activities) {
+          if (ignoredPackages.indexOf(currentInfo.activityInfo.packageName) == -1) {
+            Intent targetIntent = new Intent(intent);
+            targetIntent.setPackage(currentInfo.activityInfo.packageName);
+            targetIntents.add(targetIntent);
+          }
+        }
+      }
+
+      if (!showChooser) {
+        String packageToLaunch;
+        try {
+          if (ignoredPackages != null) {
+            packageToLaunch = targetIntents.get(0).getPackage();
+          } else {
+            packageToLaunch = activities.get(0).activityInfo.packageName;
+          }
+          intent.setPackage(packageToLaunch);
+          Log.v(TAG, "Resolve this intent with package: " + packageToLaunch);
+        } catch (IndexOutOfBoundsException exception) {
```

Comment:
I think I would suggest checking the size explicitly before accessing an item and then voiding the try/catch altogether.